### PR TITLE
Feature/watchlist

### PR DESCRIPTION
**Pull request by MStanoevich**

### CODE REVIEW

File: movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/controller/ContentController.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  @PostMapping("/addMovie")
+  public ResponseEntity<Object> addMovie(@RequestBody Movie movie) {
+    return new ResponseEntity<>(movieService.addMovie(movie), HttpStatus.OK);
+  }
+
+  @GetMapping("/movie")
+  public ResponseEntity<Object> getMovie(@RequestParam(name = "name") String name) {
+    Movie movie = movieService.getMovie(name);
+    if(movie != null){
```
Comment (slychops) [inline movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/controller/ContentController.java:38]: shouldn't be handled by the controller. should be handled by the service and return a Movie is okay, or throw exception if not okay, and the ControllerAdvice deals with the thrown exception

File: movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/controller/ContentController.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+  @PostMapping("/addSeries")
+  public ResponseEntity<Object> addSeries(@RequestBody Series series) {
+    return new ResponseEntity<>(new ResponseMovieApp(Arrays.asList(seriesService.addSeries(series))), HttpStatus.OK);
+  }
+
+  @GetMapping("/series")
+  public ResponseEntity<Object> getSeries(@RequestParam(name = "name") String name) {
+    Series series = seriesService.getSeries(name);
```
Comment (slychops) [inline movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/controller/ContentController.java:52]: same thing as for movies

File: movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/MovieServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  @Override
+  public String addMovie(Movie movie) {
+    Movie addedMovie = movieRepository.save(movie);
+
+    return addedMovie.getName() + " has been added";
+  }
+
+  @Override
+  public Movie getMovie(String name) {
+    return movieRepository.findMovieByName(name).orElse(null);
```
Comment (slychops) [inline movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/MovieServiceImpl.java:25]: better to do orElseThrow( () -> SomeException)

File: movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/SeriesServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  @Override
+  public String addSeries(Series series) {
+    Series addedSeries = seriesRepository.save(series);
+
+    return addedSeries.getName() + "  has been added";
+  }
+
+  @Override
+  public Series getSeries(String name) {
+    return seriesRepository.findSeriesByName(name).orElse(null);
```
Comment (slychops) [inline movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/SeriesServiceImpl.java:25]: better to do orElseThrow( () -> SomeException)

File: movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/WatchlistService.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.mobiquity.movieReviewApp.service;
+
+public interface WatchlistService {
+
+  String addMovieToUserWatchlist(String emailId, String movieName);
```
Comment (slychops) [inline movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/WatchlistService.java:5]: maybe change the names of this to addMovie(), removeMovie(), addSeries(), removeSeries(). I think it's redundant adding the WatchList at the end and it makes it less readable to me.

**Review by slychops — APPROVED** [review]
Approving this.. Comments are added just for you to check. 

Since this is internal I think that we don't need to have perfect code, and that we can refactor at specific times. If we refactor constantly we won't get anywhere.

File: movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/WatchlistService.java (hunk 5 above)
Comment (MStanoevich) [reply movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/WatchlistService.java:5]: good point

File: movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/controller/ContentController.java (hunk 1 above)
Comment (Supraja-Veluru) [reply movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/controller/ContentController.java:38]: same comment from me too. better handle in service.

**Review by Supraja-Veluru — APPROVED** [review]

**Pull request merged**